### ISSUE TITLE
Question about Handler and long running tasks

### ISSUE THREAD

**Issue by stephanblaurock** [unosquare/embedio#102]
since i have performance issues i have a question:
What about long running tasks? Do i see this correctly: If a request is processed, then all other requests have to wait until the first one is finished? Can the requests not be processed in parallel?

Here my code:
```c#
public class MyHandler : WebModuleBase {
		public MyHandler() {
			AddHandler("/myhandler", Unosquare.Labs.EmbedIO.Constants.HttpVerbs.Any, (context, ct) => {
				// here process a long running task
				return Task.FromResult(true);
			});
		}
		public override string Name => nameof(MyHandler).Humanize();
	}
```

**Comment by geoperez** [comment]
The WebServer creates a new Task for every incoming request. Are you sure that you aren't blocking any I/O in your long running task? Like a file or db?

**Comment by stephanblaurock** [comment]
OK, thanks for the clarification! I'll make several tests and look deeper in my code if i blocking any I/O.

**Comment by stephanblaurock** [comment]
I just made a little test: 
I used an handler like the code above and generate a long running task by Thread.Sleep(5000).
Now i opened 2 Browser with the same url. Now i reload the page on both browsers at the same time. The response on the first browser took 5s and the response on the second browser took 10s.

Now i found out that the two handler called by the same ThreadID.
Can you explain me why the two different request are handled by the same Thread?

**Comment by geoperez** [comment]
That's because a Task can share a thread with another Task, the internal scheduler will determinate if a new thread is required or not. And since you are using a thread blocking instruction (Sleep) that's is why the second request is blocked.

**Comment by geoperez** [comment]
Did you figure out why your request are blocking?

**Comment by MarcoTheFirst** [comment]
Try starting another Task in your incoming request handler, but make sure to use `TaskCreationOptions.LongRunning` in your inner task. This should force .NET to use a separate thread for the long running operation.

Example see: https://stackoverflow.com/questions/7913034/best-practice-longrunning-task-creation

**Comment by geoperez** [comment]
Hi @stephanblaurock, can I close this issue?

**Comment by mariodivece** [comment]
There was no further feedback in 10 days. Closing issue.
